plugins/default/lib/mongodb_health.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+import argparse
+import sys
+import traceback
+from collections import Counter
+
+import openshift
+import nagios
+from mongodb import ReplsetStatuses
+
+
+def generate_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-s', '--selector', type=str, action="store",
+                        required=True, help="Label Selector")
+    return parser
+
+
+check_mongodb_cmd = ("bash", "-c",
+                     "mongo 127.0.0.1/admin "
+                     "-u \"admin\" -p \"$MONGODB_ADMIN_PASSWORD\" "
+                     "--eval=\"rs.status().myState\" --quiet")
+
+
+def parse_mongo_result(output):
+    try:
+        return int(output.rstrip("\n"))
+    except:
+        return None
```

Comment:
I need a way to differentiate when the command doesn't return a replica set status code.  So anything non numeric is being converted to a None.  If we weren't breaking up the parse, analyze, and report, it would be easier to handle the special cases directly.  I could return something else like -1, but I don't like the idea of creating a new number with special meaning for somebody else's implementation.

Our current methodology for running exec in multiple pods in a single call to openshift.exec_in_pods hides our ability to track the individual return codes from each execution.